src/app/app.go

```diff
@@ -19,55 +19,60 @@ type Application struct {
 }
 
 // TestConnectToMaster check if master exists and get assigned id
-func (a *Application) TestConnectToMaster(testIP string) bool {
-	masterURL := "http://" + testIP
-	parsedURL, err := url.Parse(masterURL)
-	fmt.Println("[test] Test Url: " + parsedURL.String())
-	resp, err := http.Get(parsedURL.String())
-
+func (a *Application) TestConnectToMaster(URL string) error {
+	parsedURL, err := url.Parse("http://" + URL)
 	if err != nil {
-		if _, ok := err.(net.Error); ok {
-			fmt.Println("[test] Couldn't connect to cluster.", a.Me.NodeID)
-			fmt.Println(err)
-		}
-	} else {
-		fmt.Println(resp)
-		return true
+		log.Println("Failed to Parse URL")
+		return err
 	}
-	return false
+	_, err = http.Get(parsedURL.String())
+	return err
 }
 
 // JoinNetwork check if master exists and get assigned id and port
-func (a *Application) JoinNetwork(testIP string) (bool, nodecluster.NodeInfo) {
-	masterURL := "http://" + testIP + "/join_network"
-	parsedURL, err := url.Parse(masterURL)
-	fmt.Println("[test] Test Url: " + parsedURL.String())
-	resp, err := http.Get(parsedURL.String())
-
+func (a *Application) JoinNetwork(URL string, node nodecluster.NodeInfo) error {
+	parsedURL, err := url.Parse("http://" + URL + "/join_network")
+	log.Println("[test] Test Url: " + parsedURL.String())
+	msg := nodecluster.JoinNetworkMessage{
+		Node: a.Me,
+	}
+	body, err := json.Marshal(msg)
 	if err != nil {
-		if _, ok := err.(net.Error); ok {
-			fmt.Println("[test] Couldn't connect to cluster.", a.Me.NodeID)
-			fmt.Println(err)
-		}
-	} else {
-		fmt.Println("[test] Connected to cluster. Sending message to node.")
+		log.Println("Failed to create json message body")
+		return err
+	}
+	resp, err := http.Post(parsedURL.String(), "application/json", bytes.NewBuffer(body))
 
-		defer resp.Body.Close()
+	if err != nil {
+		log.Println("[test] Couldn't connect to master.", a.Me.NodeID)
+		log.Println(err)
+		return err
+	}
+	log.Println("Connected to master. Sending message to node.")
 
-		decoder := json.NewDecoder(resp.Body)
-		var t nodecluster.AddToClusterMessage
-		err = decoder.Decode(&t)
-		if err != nil {
-			panic(err)
-		}
-		a.Me = t.Dest
-		a.Cluster = t.Cluster
+	defer resp.Body.Close()
 
-		a.Cluster.PrintClusterInfo()
+	decoder := json.NewDecoder(resp.Body)
+	var t nodecluster.AddToClusterMessage
+	err = decoder.Decode(&t)
+	if err != nil {
+		log.Println("Failed to decode response from Master Node")
+		log.Println(err)
+		return err
+	}
+	//Update self with data from the master
+	a.Me.NodeID = t.Dest.NodeID
 
-		_ = a.Cluster.SendMessageToAllNodes("/", t)
+	a.Cluster = t.Cluster
+	a.Cluster.PrintClusterInfo()
 
-		return true, t.Dest
+	var exclusions []nodecluster.NodeInfo
+	exclusions = append(exclusions, t.Dest)
```

Comment:
This can be something like exclusions := []nodecluster.NodeInfo{t.Dest}

Reply:
updated